staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -1073,33 +1059,25 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 		select {
 		case firstRequestInnerHandlerWriteErr := <-firstReqHandlerErrCh:
 			if firstRequestInnerHandlerWriteErr != http.ErrHandlerTimeout {
-				t.Fatalf("Expected error: %#v, but got: %s", http.ErrHandlerTimeout, fmtError(firstRequestInnerHandlerWriteErr))
+				t.Errorf("Expected error: %#v, but got: %s", http.ErrHandlerTimeout, fmtError(firstRequestInnerHandlerWriteErr))
 			}
 		case <-time.After(wait.ForeverTestTimeout):
 			t.Errorf("Expected the server handler to have completed: %q", firstRequestTimesOutPath)
 		}
 
-		// first request is expected to time out.
-		if isStreamReset(firstReqResult.err) || firstReqResult.response.StatusCode != http.StatusGatewayTimeout {
-			// got what was expected
-		} else if firstReqResult.err != nil {
-			t.Fatalf("Expected request: %q to get a response or stream reset, but got error: %s", firstRequestTimesOutPath, fmtError(firstReqResult.err))
-		} else if firstReqResult.response.StatusCode != http.StatusGatewayTimeout {
-			t.Errorf("Expected HTTP status code: %d for request: %q, but got: %#+v", http.StatusGatewayTimeout, firstRequestTimesOutPath, firstReqResult.response)
-		}
+		// first request is expected to time out with
+		// an http.StatusGatewayTimeout, not with a stream reset error.
+		expectStatusCode(t, http.StatusGatewayTimeout, firstRequestTimesOutPath, firstReqResult.response, firstReqResult.err)
 
-		// second request is expected to either be rejected (ideal behavior) or time out (current approximation of the ideal behavior)
 		secondReqResult := <-secondReqResultCh
 		if isClientTimeout(secondReqResult.err) {
 			t.Fatalf("the client has unexpectedly timed out - request: %q error: %s", secondRequestEnqueuedPath, fmtError(secondReqResult.err))
 		}
-		if isStreamReset(secondReqResult.err) || secondReqResult.response.StatusCode == http.StatusTooManyRequests || secondReqResult.response.StatusCode == http.StatusGatewayTimeout {
-			// got what was expected
-		} else if secondReqResult.err != nil {
-			t.Fatalf("Expected request: %q to get a response or stream reset, but got error: %s", secondRequestEnqueuedPath, fmtError(secondReqResult.err))
-		} else if !(secondReqResult.response.StatusCode == http.StatusTooManyRequests || secondReqResult.response.StatusCode == http.StatusGatewayTimeout) {
-			t.Errorf("Expected HTTP status code: %d or %d for request: %q, but got: %#+v", http.StatusTooManyRequests, http.StatusGatewayTimeout, secondRequestEnqueuedPath, secondReqResult.response)
-		}
+		// second request is expected to be rejected by APF with http.StatusTooManyRequests
+		// a) we don't expect the timeout filter to time it out with http.StatusGatewayTimeout
+		// b) we don't expect a stream reset error, since the second request
```

Comment:
The timeout behavior is currently still being provided by the Kube's timeout filter (handler), right? If so, does this change make this test flaky until the timeout provider is also changed?

Reply:
we recently improved the apf queue timeout 
https://github.com/kubernetes/kubernetes/blob/3a4c35cc89c0ce132f8f5962ce4b9a48fae77873/staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go#L400-L419

The timeout of the second request is `5s`,  so, the second request is expected to be rejected after `1.25s`, and the timeout filter is expected to send a timeout response after `5s`

Do you think this still could be flaky? I did a `stress` on this test, and it was fine. we can increase the request timeout to `10s` to increase the delay between these two events to `7.75s`, Ideally I would like to check for `http.StatusTooManyRequests` only to since this is the behavior we are expecting here.

Reply:
do we need a change to the comment above:

```
		// In the scenario tested here, there is thus a race between two goroutines to respond to
		// the second request and any of their responses is allowed by the test.
```

Reply:
> I did a `stress` on this test, and it was fine

did you stress with a binary built with `-race`?

Reply:
> do we need a change to the comment above:

Yes, I updated the comment